test/e2e/fishapp/dynamic_stack.go

```diff
@@ -0,0 +1,811 @@
+		namespace, err = f.K8sClient.CoreV1().Namespaces().Patch(namespace.Name, types.StrategicMergePatchType, patch)
+		Expect(err).NotTo(HaveOccurred())
+		s.namespace = namespace
+	})
+}
+
+func (s *DynamicStack) deleteMeshAndNamespace(ctx context.Context, f *framework.Framework) []error {
+	var deletionErrors []error
+	if s.namespace != nil {
+		By(fmt.Sprintf("delete namespace: %s", s.namespace.Name), func() {
+			foregroundDeletion := metav1.DeletePropagationForeground
+			if err := f.K8sClient.CoreV1().Namespaces().Delete(s.namespace.Name, &metav1.DeleteOptions{
+				GracePeriodSeconds: aws.Int64(0),
+				PropagationPolicy:  &foregroundDeletion,
+			}); err != nil {
+				utils.Logf("failed to delete namespace: %s due to %v", s.namespace.Name, err)
+				deletionErrors = append(deletionErrors, err)
+			}
+		})
+	}
+	if s.mesh != nil {
+		By(fmt.Sprintf("delete mesh %s", s.mesh.Name), func() {
+			foregroundDeletion := metav1.DeletePropagationForeground
+			if err := f.K8sMeshClient.AppmeshV1beta1().Meshes().Delete(s.mesh.Name, &metav1.DeleteOptions{
+				GracePeriodSeconds: aws.Int64(0),
+				PropagationPolicy:  &foregroundDeletion,
+			}); err != nil {
+				utils.Logf("failed to delete mesh: %s due to %v", s.mesh.Name, err)
+				deletionErrors = append(deletionErrors, err)
+			}
+		})
+	}
+	return deletionErrors
+}
+
+func (s *DynamicStack) createCloudMapNamespace(ctx context.Context, f *framework.Framework) {
+	cmNamespace := fmt.Sprintf("%s-%s", f.Options.ClusterName, utils.RandomDNS1123Label(6))
+	By(fmt.Sprintf("create cloudMap namespace %s", cmNamespace), func() {
+		resp, err := f.SDClient.CreatePrivateDnsNamespaceWithContext(ctx, &servicediscovery.CreatePrivateDnsNamespaceInput{
+			Name: aws.String(cmNamespace),
+			Vpc:  aws.String(f.Options.AWSVPCID),
+		})
+		Expect(err).NotTo(HaveOccurred())
+		s.cloudMapNamespace = cmNamespace
+		utils.Logf("created cloudMap namespace %s with operationID %s", cmNamespace, aws.StringValue(resp.OperationId))
+	})
+}
+
+func (s *DynamicStack) deleteCloudMapNamespace(ctx context.Context, f *framework.Framework) []error {
+	var deletionErrors []error
+	if s.cloudMapNamespace != "" {
+		By(fmt.Sprintf("delete cloudMap namespace %s", s.cloudMapNamespace), func() {
+			var cmNamespaceID string
+			f.SDClient.ListNamespacesPagesWithContext(ctx, &servicediscovery.ListNamespacesInput{}, func(output *servicediscovery.ListNamespacesOutput, b bool) bool {
+				for _, ns := range output.Namespaces {
+					if aws.StringValue(ns.Name) == s.cloudMapNamespace {
+						cmNamespaceID = aws.StringValue(ns.Id)
+						return true
+					}
+				}
+				return false
+			})
+			if cmNamespaceID == "" {
+				err := errors.Errorf("cannot find cloudMap namespace with name %s", s.cloudMapNamespace)
+				utils.Logf("failed to delete cloudMap namespace: %s due to %v", s.cloudMapNamespace, err)
+				deletionErrors = append(deletionErrors, err)
+				return
+			}
+
+			// hummm, let's fix the controller bug in test cases first xD:
```

Comment:
Can you link to the controller bug you are fixing here please?

Reply:
thanks, added the bug links into comments :D 
```
// https://github.com/aws/aws-app-mesh-controller-for-k8s/issues/107
// https://github.com/aws/aws-app-mesh-controller-for-k8s/issues/131
```